website/content/en/docs/about/under-the-hood/architecture/buffering-model.md

```diff
@@ -0,0 +1,254 @@
+
+Part of Vector's topology design involves propagating _backpressure_, which is a signal that events cannot be
+processed as quickly as they are being received. When one component tries to send more events to a
+component than that component can currently handle, the sending component is informed of this
+indirectly. Backpressure can travel all the way from a sink, up through transforms, back to a
+source, and ultimately, even to clients such as applications sending logs over HTTP.
+
+Backpressure is a means of allowing a system to expose whether or not it can handle more work or if
+it is too busy to do so. We rely on backpressure to be able to slow down the consumption or
+acceptance of events, such as when pulling them from a source like Kafka, or accepting them over a
+socket like HTTP.
+
+In some cases, though, we don't always want to immediately propagate backpressure, as this could
+lead to constantly slowing down upstream components and callers, potentially causing issues outside
+of Vector. We want to avoid entirely slowing things down when a component just crosses over the
+threshold of being fully saturated, as well being able to handle temporary slowdowns and outages in external
+services that sinks send data to.
+
+Buffering is the approach that Vector takes to solve these problems.
+
+## Buffering between components
+
+All components in a Vector topology have a small in-memory buffer between them. The primary purpose
+of this buffer is act as the channel that two components communicate over, but we take this a little
+further by ensuring that there is a small amount of space -- typically 100 events -- that can be
+used to send events even if the component on the receiving end is currently busy. This allows
+maximizing throughput when workloads are not entirely uniform.
+
+However, in order to provide protection against temporary overloads or outages, we need to provide a
+more comprehensive buffering solution that can be tailored for the given workload.
+
+## Buffering at the sink
+
+When working with a Vector configuration, you'll be working with buffer configuration settings on
+sinks. The main reason for this is that, in practice, sinks represent the primary source of
+backpressure in a topology: talking to services over the network, where latency may be introduced,
+or outages may temporarily occur.
+
+By default, sinks use an in-memory buffer like all other components do, but the default buffer size
+is slightly increased, at 500 events. We've increased the buffer capacity for sinks specifically as,
+again, they are typically the primary source of backpressure in any given Vector topology.
+
+Beyond the default buffer capacity being larger, you can also fully control the buffer configuration
+as well.  Vector exposes two main settings for controlling buffering: the type of buffer to use, and
+the action to take when the buffer is full.
+
+## Buffer types
+
+### In-memory buffers
```

Comment:
Is it worth a specific note in this section the size of events being internal representation not the serialized event?

Reply:
I added a blurb in b6d0f1d54 that mentions the internal representation vs serialized/compressed portion, and gives a rule of thumb about estimation using the JSON serialized size, sans compression.